sdk/servicebus/azure-messaging-servicebus/README.md

```diff
@@ -292,6 +292,15 @@ between clients which can be achieved by sharing the top level builder as shown
         .buildClient();
 }
 ```
+### Receiver V/S Processor
+ When to use  'ServiceBusProcessorClient', 'ServiceBusReceiverClient' or ServiceBusReceiverAsyncClient? The processor 
+ is built using 'ServiceBusReceiverAsyncClient', it provides a convenient way of receiving message with default 
+ auto complete and auto renew of message locks in 'PEEK_LOCK' mode. The processor makes more sense where the 
+ applications have not made complete move to async client but want to process message in parallel. The 
+ 'ServiceBusProcessorClient:processMessage()' function call is made for each message in parallel.
```

Comment:
My understanding the processor client is not mainly for the case of processing messages in parallel. By default the concurrency is 1. It's a sync high-level API that is easier to use than the low-level sync API. It's trying to recover from errors and receiving for ever. The user just need to pass in a callback.